subsys/logging/Kconfig.misc

```diff
@@ -94,6 +94,15 @@ config LOG_DOMAIN_NAME
 	string "Domain name"
 	default ""
 
+config LOG_CUSTOM_HEADER
```

Comment:
There's `subsys/logging/Kconfig.mode::CONFIG_LOG_FRONTEND` I think we should:
1. Put this config in the same `Kconfig.mode`
2. Figure out how to document the differences.

It's a shame that this `CONFIG_LOG_FRONTEND` wasn't done at the macro level. Maybe there's a way to change that? I couldn't find one, but maybe someone more familiar with the logging subsystem knows?

Reply:
Moved to Konfig.mode. Updated CONFIG_LOG_FRONTEND to mention LOG_CUSTOM_HEADER.